### PR TITLE
Add support for special ayat

### PR DESCRIPTION
**Pull request by fdalvi**
This PR adds support for special ayat such as `اعوذ بالله من الشيطان الرجيم` and `بسم الله الرحمن الرحيم`. These ayat are detected, but ignored by the follow along algorithm as they are commonly recited at the beginning of sessions regardless of the actual surah. No events are fired for these ayat.

### CODE REVIEW

File: config/transcriptionConstants.js [hunk 1]
```diff
@@ -2,6 +2,10 @@
 export const MIN_PARTIAL_LENGTH = 6;
 export const MAX_IQRA_MATCHES = 10;
 export const PREFIX_MATCHING_SLACK = 5;
-export const QUERY_PREFIX_FRACTION = 0.50;
+export const QUERY_PREFIX_FRACTION = 0.30;
```
Comment (Ananasship) [inline config/transcriptionConstants.js]: To clarify: this isn't related to special ayahs but improves the accuracy, correct?

File: config/transcriptionConstants.js [hunk 2]
```diff
@@ -2,6 +2,10 @@
...
 export const MIN_PARTIAL_LENGTH_FRACTION = 0.05;
-export const TRANSCRIPTION_SLACK = 10;
\ No newline at end of file
+export const TRANSCRIPTION_SLACK = 10;
+export const SPECIAL_AYAT = [
```
Comment (Ananasship) [inline config/transcriptionConstants.js]: lgtm, in the future we can move the list of Ayahs in a separate file and a small comment explaining what special ayahs is used for (might not be understood what special means just from the name)

Comment (Ananasship) [reply config/transcriptionConstants.js]: Not sure if `SPECIAL_AYAT` is going to be used for ayahs we're ignoring in particular. Would be nice to clarify that. 

If it's the former case, perhaps we can rename it to something like `TEXT_TO_IGNORE`

File: src/transcribe.js [hunk 3]
```diff
@@ -75,21 +94,44 @@ export class Transcriber {
...
-                        {'surahNum': surahNum, 'ayahNum': ayahNum+1}
-                    ];
+                    // Check for special ayat
+                    if (isSpecialAyah(matches[0])) {
+                        // Detected ayah is a special ayah, so just perform follow along until we
+                        // reach the end of it. We do not have a next ayah because we do not know what
+                        // surah we are in.
+                        // Note: No ayahFound event is fired for special ayat
+                        if (process.env.DEBUG === 'development') {
+                            console.log(`[Follow along] Special Ayah Detected: ${matches[0].text_simple}`);
```
Comment (Ananasship) [inline src/transcribe.js]: Nice! thanks for the log

**Comment by piraka9011** [comment]
Confirming this works.
See the log below:
```
[M25vz6RCv0Wfdp_rAAAC] Partial result:  بسم
[Follow Along] Iqra query is:   بسم
[Follow along] Current Matches: []
[M25vz6RCv0Wfdp_rAAAC] Partial result:  بسم الله
[Follow Along] Iqra query is:   بسم الله
[Follow along] matching بسم الله with وقال اركبوا فيها بسم الله مجراها ومرساها ان ربي لغفور رحيم UNMATCHED
[Follow along] matching بسم الله with انه من سليمان وانه بسم الله الرحمن الرحيم UNMATCHED
[Follow along] matching بسم الله with اعوذ بالله من الشيطان الرجيم UNMATCHED
[Follow along] matching بسم الله with بسم الله الرحمن الرحيم MATCHED
[Follow along] Current Matches: [{"chapter_id":-1,"verse_number":1,"text_simple":"بسم الله الرحمن الرحيم"}]
[Follow along] Special Ayah Detected: بسم الله الرحمن الرحيم
[Follow along] Partial transcript:  اعوذ بالله من الشيطان الرجيم  بسم الله
[Follow along] Correct ayah: بسم الله الرحمن الرحيم
[M25vz6RCv0Wfdp_rAAAC] Partial result:  بسم
[Follow along] Partial transcript:  اعوذ بالله من الشيطان الرجيم  بسم
[Follow along] Correct ayah: بسم الله الرحمن الرحيم
[M25vz6RCv0Wfdp_rAAAC] Partial result:  بسم
[Follow along] Partial transcript:  اعوذ بالله من الشيطان الرجيم  بسم
[Follow along] Correct ayah: بسم الله الرحمن الرحيم
[M25vz6RCv0Wfdp_rAAAC] Partial result:  بسم الله
[Follow along] Partial transcript:  اعوذ بالله من الشيطان الرجيم  بسم الله
[Follow along] Correct ayah: بسم الله الرحمن الرحيم
[M25vz6RCv0Wfdp_rAAAC] Partial result:  بسم الله
[Follow along] Partial transcript:  اعوذ بالله من الشيطان الرجيم  بسم الله
[Follow along] Correct ayah: بسم الله الرحمن الرحيم
[M25vz6RCv0Wfdp_rAAAC] Partial result:  بسم الله الرحمن
[Follow along] Partial transcript:  اعوذ بالله من الشيطان الرجيم  بسم الله الرحمن
[Follow along] Correct ayah: بسم الله الرحمن الرحيم
[M25vz6RCv0Wfdp_rAAAC] Partial result:  بسم الله الرحمن الرحيم
[Follow along] Partial transcript:  اعوذ بالله من الشيطان الرجيم  بسم الله الرحمن الرحيم
[Follow along] Correct ayah: بسم الله الرحمن الرحيم
[Follow along] Detected end of ayah
[M25vz6RCv0Wfdp_rAAAC] Final result:  بسم الله الرحمن الرحيم
[Follow Along] Iqra query is: م
[Follow along] Current Matches: []
```

File: config/transcriptionConstants.js (hunk 1 above)
Comment (fdalvi) [reply config/transcriptionConstants.js]: Ahh yes this was me fiddling with the algorithm and it slipped into the commit, Nice catch!

File: config/transcriptionConstants.js (hunk 2 above)
Comment (fdalvi) [reply config/transcriptionConstants.js]: Right, I see what you mean. In my mind these are all ayat (or pseudo-ayat like `اعوذ بالله من الشيطان الرجيم` that we want to ignore specifically in the first matching phase (where we are trying to find where we are in the Quran). Other ayat that I potentially envision going into this list are repeated ayat like in Surah Rahman - where if the user recites `فبأي آلاء ربكما تكذبان` we ignore that in the matching phase and wait for the next ayah to be sure before displaying the text to the user and continuing follow-along.

**Pull request merged**